forward/forward.go

```diff
@@ -66,3 +68,99 @@ func Wrap(msg stanza.Message, body string, received time.Time, r xml.TokenReader
 		}.Wrap(r),
 	))
 }
+
+// tokenBuffer stores tokens inside a slice and implements xml.TokenReader & xmlstream.TokenWriter.
+// Useful when we need to copy a stream and then read it later
```

Comment:
Unless there is a better way 🤔. I could have used xmlstream.ReadAll and use an xmlstream.ReaderFunc, or maybe encode the stream with an encoder and decode it again (which seems an overkill)? Or maybe something else?

Reply:
Humm, I see what you mean. In this case, the user should not assume that the delay is unmarshaled until she has consumed the entire returned stream.  Will give it a try.

Reply:
What about this version? Is this clean enough, or maybe I should read tokens directly and use the token level to decide about delay unmarshaling?

Reply:
If I understood what you meant by "read tokens directly", I think that's the correct way with the least overhead. I've left some comments inline that I hope will make things clearer. Thanks again!